### PR TITLE
update to 2018 edition

### PR DESCRIPTION
**Pull request by Johann150**

### CODE REVIEW

File: src/theca/utils.rs [hunk 1]
```diff
@@ -374,7 +380,14 @@ pub fn parse_last_touched(lt: &str) -> Result<OffsetDateTime> {
...
     let t = parse_last_touched(lt)?;
-    Ok(t.to_offset(UtcOffset::local_offset_at(t)).format(DATEFMT_SHORT))
+    Ok(t.to_offset(
+        UtcOffset::try_local_offset_at(t)
+            // at least try the current offset
+            .or_else(|_| UtcOffset::try_current_local_offset())
+            // give up and assume UTC
+            .unwrap_or_else(|_| UtcOffset::hours(0)),
+    )
+    .format(DATEFMT_SHORT))
```
Comment (Johann150) [inline src/theca/utils.rs:390]: There is still a problem with this that I don't know how to resolve: The program gets the time zone offset even in tests. This means that the tests will fail depending on location of the machine the tests are run on. The tests will only succeed if the machine is itself in the `-0800` pacific time zone (or the DST variant thereof). Instead of this code, something like this might solve it. But this way there would of course be some edge cases missed, e.g. summer time, but with a fixed date that is probably not relevant anyway?
```suggestion
    #[cfg(not(test))]
    let offset = UtcOffset::try_local_offset_at(t)
            // at least try the current offset
            .or_else(|_| UtcOffset::try_current_local_offset())
            // give up and assume UTC
            .unwrap_or_else(|_| UtcOffset::hours(0));
    #[cfg(test)] // pin time zone to UTC
    let offset = UtcOffset::hours(0);
    Ok(t.to_offset(offset).format(DATEFMT_SHORT))
```
The tests would have to be adjusted of course.
